README.md

```diff
@@ -34,51 +39,116 @@ We verify this package against N-2 recent versions of Go compiler. As of Septemb
+[Environment](https://docs.sentry.io/enriching-error-data/environments/?platform=go)
+are read from the environment variables `SENTRY_DSN`, `SENTRY_RELEASE` and
+`SENTRY_ENVIRONMENT`, respectively.
+
+More on this in the [Configuration](https://docs.sentry.io/platforms/go/config/)
+section of the official Sentry documentation.
 
 ## Usage
 
-By default, Sentry Go SDK uses asynchronous transport, which in the code example below requires an explicit awaiting for event delivery to be finished using `sentry.Flush` method. It is necessary, because otherwise the program would not wait for the async HTTP calls to return a response, and exit the process immediately when it reached the end of the `main` function. It would not be required inside a running goroutine or if you would use `HTTPSyncTransport`, which you can read about in `Transports` section.
+The SDK must be initialized with a call to `sentry.Init`. The default transport
+is asynchronous and thus most programs should call `sentry.Flush` to wait until
+buffered events are sent to Sentry right before the program terminates.
+
+Typically, `sentry.Init` is called in the beginning of `func main` and
+`sentry.Flush` is [deferred](https://golang.org/ref/spec#Defer_statements) right
+after.
+
+> Note that if the program terminates with a call to
+> [`os.Exit`](https://golang.org/pkg/os/#Exit), either directly or indirectly
+> via another function like `log.Fatal`, deferred functions are not run.
+>
+> In that case, and if it is important for you to report outstanding events
+> before terminating the program, arrange for `sentry.Flush` to be called before
+> the program terminates.
+
+Example:
 
 ```go
+// This is an example program that makes an HTTP request and prints response
+// headers. Whenever a request fails the error is reported to Sentry.
+//
+// Save this into a file named "main.go".
+//
+// Try it by running:
+//
+// 	go run main.go
+// 	go run main.go https://sentry.io
+// 	go run main.go bad-url
+//
+// To actually report events to Sentry, set the DSN either by editing the
+// appropriate line below, or set the environment variable
+// SENTRY_DSN=<your-dsn>. Replace "<your-dsn>" with a proper DSN you can find in
+// your Sentry project settings.
 package main
 
 import (
-    "fmt"
-    "os"
-    "time"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
 
-    "github.com/getsentry/sentry-go"
+	"github.com/getsentry/sentry-go"
 )
 
 func main() {
-  err := sentry.Init(sentry.ClientOptions{
-    Dsn: "___DSN___",
-  })
-
-  if err != nil {
-    fmt.Printf("Sentry initialization failed: %v\n", err)
-  }
-  
-  f, err := os.Open("filename.ext")
-  if err != nil {
-    sentry.CaptureException(err)
-    sentry.Flush(time.Second * 5)
-  }
+	if len(os.Args) < 2 {
+		log.Fatalf("usage: %s URL", os.Args[0])
+	}
+
+	err := sentry.Init(sentry.ClientOptions{
+		Dsn:   "",   // set DSN here or set SENTRY_DSN environment variable
+		Debug: true, // enabling debug output can be helpful at times
```

Comment:
`debub=true` can be less obvious, could be worth pointing out what is actually does:

```suggestion
		Debug: true, // to see debug messages of the SDK itself
```

Reply:
I had a more verbose version of this in a side branch

```suggestion
               // Either set your DSN here or set the environment variable SENTRY_DSN
               // and omit this field altogether.                                    
               Dsn: "",                                                              
               // Set Debug to true if you want the SDK to output debug messages. Can
               // be useful when you're getting started.                             
               //Debug: true,                                                        
```

Will update, thanks for noticing!